### PR TITLE
feat(sealevel): Address Lookup Table support for reduced transaction size

### PR DESCRIPTION
**Pull request by tkporter**
## Problem

Aleo→Solana warp route messages are exceeding Solana's 1232-byte transaction size limit by ~9 bytes. This is due to the larger message bodies from Aleo (256 bytes vs typical 64 bytes for EVM chains).

**Root cause**: Solana transactions include the full 32-byte pubkey for every account referenced. Warp route `process()` transactions require ~13-15 accounts, consuming significant space before the instruction data.

Example of an Aleo message that failed delivery (9 bytes over limit):
```
Account Metas (19 total):
  [0] 6DjHX6Ezjpq3zZMZ8KsqyoFYo1zPSDoiZmLLkxD4xKXS (signer, readonly) - Payer
  [1] 11111111111111111111111111111111 (readonly) - System Program
  [2] H3EgdESu59M4hn5wrbeyi9VjmFiLYM7iUAbGtrA5uHNE (writable) - Inbox PDA
  [3] 3knbm4ytq6eMA3fqh1yf8c6NVZcx9W1XPvtAQNA37Tbh (readonly) - Process Authority
  [4] 9CFsBU9yBDpTwpvNfjLH6L1vv5LVKW2rMgcXfpW6ZTi8 (writable) - Processed Message
  [5] 9oD3kZDEkLfYVL7DPucCkpVBbjnQ5rrU7PMCaHAzRvkZ (readonly)
  [6] noopb9bkMVfRPU8AsbpTUg8AQkHtKwMYZiFUjNRtMmV (readonly) - SPL Noop
  [7] Da6Lp9syj8hLRiqjZLTLbZEC1NPhPMPd1JJ3HQRN4NyJ (readonly) - ISM
  [8-12] ISM verify accounts
  [13] TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA (readonly) - Token Program
  [14] ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL (readonly) - ATA Program
  [15] Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB (readonly) - USDT Mint
  [16-18] Warp route handle accounts (writable)

Transaction Size: 1241 bytes (max: 1232) ❌
```

## Solution

Implement **Address Lookup Tables (ALTs)** - a Solana feature that allows transactions to reference accounts via a 1-byte index into a pre-loaded table instead of the full 32-byte pubkey. This reduces transaction size by ~120 bytes (saving 31 bytes per compressed account).

### Real-World Results

Tested with WBTC route (Optimism → Solana collateral warp route):

| Scenario | Size | Link |
|----------|------|------|
| Without ALT | 1231 bytes (1 byte under max!) | [solscan](https://solscan.io/tx/inspector?txhash=34QsTKZhNKyKmspHq9YMaGDy58YoNVS2nppffetFNwdntRZXBiGBVhtQ8RqjXzcNTCz8qqEmiC4wkYuQZpiLMbxK) |
| With ALT | 1112 bytes | [solscan](https://solscan.io/tx/inspector?txhash=4sLz6iwYjyxugCVnYu6zX9uZm6b8FhZm53iQjTtBPEAsJ3iXExLGQqbG52zSWdEa2TrkLAdAZa8x91ZQtKy1PbWz) |

**119 bytes saved** - more than enough headroom for the 9-byte overflow from Aleo messages.

Recent versioned tx from deployed relayer (RC): [solscan](https://solscan.io/tx/5QeRA3o76MbazLdeAXMSKZStrbJnyfEetpQTjB7Qoh9KsQiYGjkywt3tj69spKkh1yDfifKb8DqD2bkhYP3ardE7)

#### 1. Relayer - Switch to VersionedTransaction (when ALT configured)

- **`provider.rs`**: Added lazy `AddressLookupTableCache` that loads ALT data on first use. The `build_estimated_tx_for_instruction()` method now builds `MessageV0` with ALT when configured, falling back to legacy `Transaction` when no ALT is configured.

- **`tx_type.rs`** (new): `SealevelTxType` enum abstracting `Legacy(Transaction)` vs `Versioned(VersionedTransaction)`. This allows the codebase to handle both formats uniformly.

- **`tx_submitter.rs`**: Updated `TransactionSubmitter` trait to use `SealevelTxType` instead of legacy `Transaction`.

- **`rpc/client.rs` & `rpc/fallback.rs`**: Added `send_versioned_transaction()` and `simulate_versioned_transaction()` methods, plus unified `send_sealevel_tx()` / `simulate_sealevel_tx()` dispatchers.

- **`mailbox.rs`**: New `SealevelProcessPayload` struct that bundles the process instruction with optional ALT address. This is returned by `process_calldata()` for use by Lander.

#### 2. CLI - ALT Creation Command

Added `hyperlane-sealevel-client alt create --mailbox <ADDR>` command that:
1. Creates a new ALT
2. Extends it with common static accounts eligible for compression:
   - System Program
   - Mailbox Inbox PDA (derived from mailbox)
   - SPL Noop (logging program)
   - Token Program (CPI'd by warp routes)
   - Token 2022 Program (CPI'd by synthetic warp routes)
   - Associated Token Account Program (CPI'd by warp routes)
3. Freezes the ALT to make it immutable

#### 3. Configuration

New optional `mailboxProcessAlt` field in chain config. When set, versioned transactions with ALT are used for message processing on that chain. When absent, legacy transactions are used.

### Breaking Changes

**Internal serialization format change**: `Mailbox::process_calldata()` now returns serialized `SealevelProcessPayload` instead of raw `Instruction`. This affects the internal contract between Mailbox and Lander. Since Lander ships as part of the relayer binary, this is not a user-facing breaking change, but developers should be aware when reviewing message processing code.

### Tradeoffs Considered

| Option | Pros | Cons | Decision |
|--------|------|------|----------|
| **ALT (chosen)** | ~120 byte reduction, no protocol changes, works with existing contracts | Operational complexity, ALT must be pre-deployed | ✅ Best balance |
| **Smaller Aleo messages** | Fixes root cause | Requires Aleo contract changes, may not be feasible | ❌ Protocol constraints |
| **Compression** | Transparent | Solana doesn't support tx compression | ❌ Not available |

| Design Choice | Decision | Rationale |
|---------------|----------|-----------|
| Shared vs per-route ALT | Shared | Most warp routes use same programs (Token, ATA). Route-specific accounts remain in tx. |
| Frozen vs mutable | Frozen | Security - prevents malicious account injection. New accounts require new ALT. |
| Config-driven | Yes | Only chains needing ALT opt-in. Maintains compatibility with all SVM chains. |
| Legacy vs Versioned without ALT | Legacy | Some SVM chains (e.g., Solaxy) don't support versioned transactions. Legacy is the safe default. |

### What's NOT in ALT

The **Mailbox program ID itself** is NOT eligible for ALT - it's the top-level `program_id` field which isn't part of the accounts array. However, programs called via CPI (Token, ATA) ARE in the accounts array and benefit from ALT.

## E2E Testing Strategy

Added a third test chain (`sealeveltest3`) to verify both transaction paths work correctly:

| Destination | ALT Configured | Transaction Type | Purpose |
|-------------|----------------|------------------|---------|
| `sealeveltest2` | Yes | `VersionedTransaction` with ALT | Tests new versioned tx path |
| `sealeveltest3` | No | Legacy `Transaction` | Tests legacy fallback path |

Messages flow from `sealeveltest1` (origin with single validator) to both destinations:
- 6 messages to `sealeveltest2` (3 before agents start, 3 after) - exercises versioned tx with ALT
- 6 messages to `sealeveltest3` (3 before agents start, 3 after) - exercises legacy tx without ALT

Provider metrics are checked separately for each destination to verify successful delivery on both paths.

## Testing

- [x] Sealevel e2e tests pass with both VersionedTransaction and legacy Transaction paths
- [x] Clippy clean
- [x] ALTs deployed and configured for Solana mainnet and testnet
- [x] Verified on RC with real warp route traffic (WBTC Optimism→Solana)

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Address Lookup Table (ALT) support and Versioned transaction handling for smaller Sealevel transactions.
  * New CLI "alt" subcommand to create/extend/freeze ALTs.
  * Unified transaction abstraction to support legacy and versioned flows across the stack.

* **Configuration**
  * Optional mailboxProcessAlt setting to enable ALT-based mailbox processing.
  * Mainnet/test configs updated to include mailboxProcessAlt entries.

* **Testing / Environments**
  * Added sealeveltest3 dev/test environment, updated local e2e flows, fixtures, and mocks to cover legacy and ALT/versioned paths.

### CODE REVIEW

File: rust/main/lander/tests/integration_sealevel.rs [hunk 1]
```diff
@@ -402,15 +422,16 @@ async fn test_sealevel_payload_estimation_failure_results_in_dropped() {
...
         .expect_get_estimated_costs_for_instruction()
-        .returning(|_, _, _, _| Err(eyre::eyre!("Estimation failed").into()));
+        .returning(|_, _, _, _, _| Err(eyre::eyre!("Estimation failed").into()));
 
     provider
         .expect_create_transaction_for_instruction()
-        .returning(|_, _, instruction, payer, _, _| {
-            Ok(SealevelTransaction::new_unsigned(Message::new(
+        .returning(|_, _, instruction, payer, _, _, _| {
+            let tx = SolanaTransaction::new_unsigned(Message::new(
```
Comment (ameten) [inline rust/main/lander/tests/integration_sealevel.rs]: Should we test it with versioned transaction as well?

File: rust/main/config/test_sealevel_config.json
[the remark this answers is not in the document]
Comment (tkporter) [reply rust/main/config/test_sealevel_config.json]: The address unfortunately relies on the authority address (the initial deployer of it, even though our tooling revokes its authority to update it) + the slot number of a recent block (the program that's invoked as part of creating the ALT requires this slot to be verifiably recent). So not really straightforward to get determinism here as I'd rather not hardcode a specific slot number to not rely on the tx landing around a specific block height

To illustrate, from their SDK:
```
pub fn derive_lookup_table_address(
    authority_address: &Pubkey,
    recent_block_slot: Slot,
) -> (Pubkey, u8) {
    Pubkey::find_program_address(
        &[authority_address.as_ref(), &recent_block_slot.to_le_bytes()],
        &id(),
    )
}
```

**Comment by tkporter** [comment]
Re: claude's feedback - imo these aren't a big deal but I did 3

**Comment by tkporter** [comment]
@ameten I believe I satisfied this
> It will be good to add tests with no-None alt table address to Lander.

With https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/7985/commits/dd00955ff8c1157b4a715e9717368c5cf43145dc, lmk what you think
